cpp/ql/src/Critical/UseAfterFree.ql

```diff
@@ -135,18 +135,24 @@ module ParameterSinks {
   }
 }
 
-predicate isUse(DataFlow::Node n, Expr e) {
-  isUse0(n, e)
-  or
-  exists(CallInstruction call, int i, InitializeParameterInstruction init |
-    n.asOperand().getDef().getUnconvertedResultExpression() = e and
-    init = ParameterSinks::getAnAlwaysDereferencedParameter() and
-    call.getArgumentOperand(i) = n.asOperand() and
-    init.hasIndex(i) and
-    init.getEnclosingFunction() = call.getStaticCallTarget()
-  )
+module IsUse {
+  private import semmle.code.cpp.ir.dataflow.internal.DataFlowImplCommon
```

Comment:
I don't particularly like that we have to pull in something internal. Is this really necessary?

Reply:
Let's not re-invent anything, but if do wonder if it's really necessary to pull this in. Why don't the introduced pragmas suffice?

Reply:
We discussed this in a call and decided to have this internal import here for now with the goal of:
1. Identifying other places where we're doing this parameter-argument mapping manually or using this internal import, and
2. Fixing this with a non-internal predicate.

Reply:
We briefly discussed this one-on-one, and pulling in the internal predicate seems the best solution at the moment without duplicating a lot of code from that internal predicate. So I'll accept this as-is.

As the original non-optimal join-order might also occur in other queries that have similar logic, we'll open an internal issue to track this and to provide a proper predicate that is not internal and efficient.